sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaRulesDefinition.java

```diff
@@ -112,44 +114,60 @@ protected void newRule(Class<?> ruleClass, NewRepository repository, JavaSonarWa
     if (rule == null) {
       throw new IllegalStateException("No rule was created for " + ruleClass + " in " + repository.key());
     }
-    String metadataKey = ruleMetadata(ruleClass, rule);
+    String rspecKey = rspecKey(ruleClass, rule);
+    RuleMetadata ruleMetadata = readRuleMetadata(rspecKey);
+    addMetadata(rule, ruleMetadata);
+    String ruleHtmlDescription = readRuleHtmlDescription(rspecKey);
+    if (ruleHtmlDescription != null) {
+      rule.setHtmlDescription(ruleHtmlDescription);
+    }
     // 'setActivatedByDefault' is used by SonarLint standalone, to define which rules will be active
-    rule.setActivatedByDefault(profile.ruleKeys.contains(ruleKey) || profile.ruleKeys.contains(metadataKey));
+    boolean activatedInProfile = profile.ruleKeys.contains(ruleKey) || profile.ruleKeys.contains(rspecKey);
+    boolean isSecurityHotspot = ruleMetadata != null && ruleMetadata.isSecurityHotspot();
+    rule.setActivatedByDefault(activatedInProfile && !isSecurityHotspot);
     rule.setTemplate(TEMPLATE_RULE_KEY.contains(ruleKey));
   }
 
-  private String ruleMetadata(Class<?> ruleClass, NewRule rule) {
-    String metadataKey = rule.key();
+  private static String rspecKey(Class<?> ruleClass, NewRule rule) {
     org.sonar.java.RspecKey rspecKeyAnnotation = AnnotationUtils.getAnnotation(ruleClass, org.sonar.java.RspecKey.class);
     if (rspecKeyAnnotation != null) {
-      metadataKey = rspecKeyAnnotation.value();
-      rule.setInternalKey(metadataKey);
+      String rspecKey = rspecKeyAnnotation.value();
+      rule.setInternalKey(rspecKey);
+      return rspecKey;
+    } else {
+      return rule.key();
     }
-    addHtmlDescription(rule, metadataKey);
-    addMetadata(rule, metadataKey);
-    return metadataKey;
   }
 
-  private void addMetadata(NewRule rule, String metadataKey) {
+  static RuleMetadata readRuleMetadata(String metadataKey) {
```

Comment:
should we add `@Nullable` for this method?

Reply:
Added